nipype/utils/filemanip.py

```diff
@@ -430,3 +431,13 @@ def write_rst_dict(info, prefix=''):
     for key, value in sorted(info.items()):
         out.append(prefix + '* ' + key + ' : ' + str(value))
     return '\n'.join(out)+'\n\n'
+
+
+def mkdir_p(path):
+    try:
+        os.makedirs(path)
```

Comment:
i don't think this is an atomic operation, and hence likely to run into the same issue. this is an instance where a lock may be required.